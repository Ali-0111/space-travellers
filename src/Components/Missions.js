const Missions = () => (
  <div>
    This is Missions page.
  </div>
);

export default Missions;
