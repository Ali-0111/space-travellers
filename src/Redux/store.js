import { configureStore, combineReducers } from '@reduxjs/toolkit';
import missionsSlice from './missions/missionsSlice';
import rocketSlice from './rockets/rocketsSlice';

const rootReducer = combineReducers({
  missionsReducer: missionsSlice.reducer,
  reservationListKey: rocketSlice.reducer,
});

const store = configureStore({
  reducer: rootReducer,
});

export default store;
