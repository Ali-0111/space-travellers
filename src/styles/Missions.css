table {
  display: flex;
  flex-direction: column;
  padding: 0 50px;
}

thead > tr {
  display: grid;
  grid-template-columns: 10% 65% 11% 14%;
}

thead > tr > th {
  border: 1px solid rgb(185, 182, 182);
  border-right: none;
  padding: 10px;
  text-align: left;
  word-wrap: break-word;
}

thead > tr > th:nth-last-child(1) {
  border-right: 1px solid rgb(185, 182, 182);
}

tbody {
  display: flex;
  flex-direction: column;
}

tbody > tr {
  display: grid;
  grid-template-columns: 10% 65% 11% 14%;
}

tbody > tr > td {
  padding: 10px;
  border-bottom: 1px solid rgb(185, 182, 182);
  border-left: 1px solid rgb(185, 182, 182);
}

tbody > tr > td:nth-last-child(1) {
  border-right: 1px solid rgb(185, 182, 182);
}

tbody > tr > td:nth-child(1) {
  font-weight: 600;
  word-wrap: break-word;
}

tbody > tr > td:nth-child(3) {
  font-size: 13px;
  text-align: center;
}

tbody > tr > td:nth-child(n+3) {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

tbody > tr:nth-child(odd) {
  background-color: #f2f2f2;
}

.notAMember {
  background-color: #6d757d;
  padding: 3px;
  color: white;
  border-radius: 5px;
}

.activeMember {
  background-color: #18a2b8;
  padding: 3px;
  color: white;
  border-radius: 5px;
}

button {
  background-color: inherit;
  border: none;
  padding: 10px;
  word-wrap: break-word;
}

.joinMission {
  color: #343a40;
  border: 1px solid #343a40;
  word-wrap: break-word;
}

.leaveMission {
  color: #ec5766;
  border: 1px solid #ec5766;
  word-wrap: break-word;
}
