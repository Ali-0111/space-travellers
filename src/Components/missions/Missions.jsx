import { useSelector, useDispatch } from 'react-redux';
import { setReservationTrue, setReservationFalse } from '../../Redux/missions/missionsSlice';
import '../../styles/Missions.css';

const Missions = () => {
  const dispatch = useDispatch();
  const { missionsArray } = useSelector((store) => (store.missionsReducer));
  // console.log(missionsArray);

  return (
    <table>
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th> </th>
        </tr>
      </thead>
      <tbody>
        {missionsArray?.map((entry) => (
          <tr key={entry.mission_id}>
            <td>{entry.mission_name}</td>
            <td>{entry.description}</td>
            {!entry.reserved && (
              <>
                <td><div className="notAMember">NOT A MEMBER</div></td>
                <td><button type="submit" className="joinMission" onClick={() => dispatch(setReservationTrue(entry.mission_id))}>Join Mission</button></td>
              </>
            )}
            {entry.reserved && (
              <>
                <td><div className="activeMember">Active Member</div></td>
                <td><button type="submit" className="leaveMission" onClick={() => dispatch(setReservationFalse(entry.mission_id))}>Leave Mission</button></td>
              </>
            )}
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export default Missions;
