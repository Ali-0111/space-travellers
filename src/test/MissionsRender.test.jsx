import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import store from '../Redux/store';

import Missions from '../Components/missions/Missions';

describe('Missions rendering Validation', () => {
  test('Missions should render with /missions url', () => {
    const treeSnapShot = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path="/missions" element={<Missions />} />
          </Routes>
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(treeSnapShot).toMatchSnapshot();
  });
});
