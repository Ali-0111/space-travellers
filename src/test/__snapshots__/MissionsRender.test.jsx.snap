// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions rendering Validation Missions should render with /missions url 1`] = `null`;
